server/mediaserver.go

```diff
@@ -465,10 +483,11 @@ func gotRTMPStreamHandler(s *LivepeerServer) func(url *url.URL, rtmpStrm stream.
 				} else if err != nil {
 					// Should be logged upstream
 				}
-				s.VideoNonceLock.Lock()
-				_, active := s.VideoNonce[rtmpStrm.GetStreamID()]
-				s.VideoNonceLock.Unlock()
+				s.connectionLock.Lock()
+				cxn, active := s.rtmpConnections[mid]
+				s.connectionLock.Unlock()
```

Comment:
Since we potentially modify the pointer `cxn.sess` on L490, should this unlock statement be after the `active` check (maybe using `defer`)?

Reply:
Yeah, that's probably safer. Fixed.

Right now, we don't really have a strong requirement for mutual exclusion of the `cxn` object (eg, `cxn` is currently unprotected elsewhere in `gotRTMPStreamHandler`) but using `defer` makes things more consistent with the rest of the code and gives us that extra bit of safety should it every come up later.